tests/testmongodb.nim

```diff
@@ -0,0 +1,219 @@
+  type
+    User {.table: "users".} = object
+      email {.unique.}: string
+      ssn: Option[int]
+      birthDate: Time
+    Publisher {.table: "publishers".} = object
+      title {.unique.}: string
+    Book {.table: "books".} = object
+      title: string
+      authorEmail {.fk: User.email, onDelete: "CASCADE".}: string
+      publisherTitle {.fk: Publisher.title.}: string
+      ratings: seq[float]
+      copyright: Copyright
+
+  # TODO proc getBookById(id: Oid): Book = withDb(Book.getOne id)
+  # TODO add and test foreign key pulls in mongo
+
+type
+  Edition {.table: "editions".} = object
+    id {.dbCol: "_id".}: Oid
+    title: string
+    book: Book
+
+dbAddCollection(Edition)
+
+suite "Creating and dropping tables, CRUD":
+  setup:
+    var 
+      user_reference_id: array[1..9, Oid]
+      publisher_reference_id: array[1..9, Oid]
+      book_reference_id: array[1..9, Oid]
+      edition_reference_id: array[1..9, Oid]
+
+    withDb:
+      createTables(force=true)
+
+      for i in 1..9:
+        var
+          user = User(
+            email: "test-$#@example.com" % $i, 
+            ssn: some i,
+            birthDate: parseTime("200$1-0$1-0$1" % $i, "yyyy-MM-dd", utc())
+          )
+          publisher = Publisher(
+            title: "Publisher $#" % $i
+          )
+          book = Book(
+            title: "Book $#" % $i, 
+            authorEmail: user.email,
+            publisherTitle: publisher.title,
+            ratings: @[4.5, 9.6, 7.0],
+            copyright: Copyright(legal_name: "XY $# Corp" % $i, year: 1990 + i)
+          )
+          edition = Edition(
+            title: "Edition $#" % $i
+          )
+
+        user.insert()
+        user_reference_id[i] = user.id
+        publisher.insert()
+        publisher_reference_id[i] = publisher.id
+        book.insert()
+        book_reference_id[i] = book.id
+        edition.book = book
+        edition.insert()
+        edition_reference_id[i] = edition.id
+
+  teardown:
+    withDb:
+      dropTables()
+
+  test "Create records":
+    withDb:
+      let
+        publishers = Publisher.getMany(100, sort = %*{"title": 1})
+        books = Book.getMany(100, sort = %*{"title": 1})
+        editions = Edition.getMany(100, sort = %*{"title": 1})
+
+      check len(publishers) == 9
+      check len(books) == 9
+      check len(editions) == 9
+
+      check publishers[3].title == "Publisher 4"
+
+      check books[5].title == "Book 6"
+      check books[5].ratings == @[4.5, 9.6, 7.0]
+      check books[5].copyright.legal_name == "XY 6 Corp"
+      check books[5].copyright.year == 1996
+
+      check editions[7].title == "Edition 8"
+      check editions[7].book == books[7]
+
+  test "Read records":
+    withDb:
+      var
+        users: seq[User] = @[]
+        # users = User().repeat 10
+        publishers = Publisher().repeat 10
+        books = Book().repeat 10
+        editions = Edition().repeat 10
+
+      users.pullMany(20, offset=5, sort = %*{"ssn": 1})

```

Comment:
What's the deal with renaming `getMany` to `pullMany`? Is it a Mongo-specific thing? IMO it's confusing to have different terminology for essentially the same action in different backends.

Reply:
I'll change it back. Just an opinion, but that one getMany alters the first parameter and the other getMany does not is a point of confusion, but backward compatibility is important also. I will back the change.